Strange error when running `repos setup` and target org does not exist

This is most likely an oversight in the GiteaAPI implementation.
```
[tmp] $ repobee -p gitea repos setup --students simon eve alice --assignments task-1 task-2 task-3 --tb
[WARNING] The Gitea plugin is in very early development, use at your own risk!
Cloning template repositories: 100%|███████████████████| 3/3 [00:01<00:00,  2.84repos/s]
Creating student teams:   0%|                                     | 0/3 [00:00<?, ?it/s]
[ERROR] string indices must be integers
[ERROR] Critical exception
Traceback (most recent call last):
  File "/home/slarse/.repobee/env/lib/python3.9/site-packages/_repobee/main.py", line 171, in _main
    _repobee.cli.dispatch.dispatch_command(
  File "/home/slarse/.repobee/env/lib/python3.9/site-packages/_repobee/cli/dispatch.py", line 56, in dispatch_command
    dispatch_table[category](args, config_file, api) or hook_results
  File "/home/slarse/.repobee/env/lib/python3.9/site-packages/_repobee/cli/dispatch.py", line 80, in _dispatch_repos_command
    return command.setup_student_repos(
  File "/home/slarse/.repobee/env/lib/python3.9/site-packages/_repobee/command/repos.py", line 83, in setup_student_repos
    platform_teams = _create_platform_teams(teams, api)
  File "/home/slarse/.repobee/env/lib/python3.9/site-packages/_repobee/command/repos.py", line 115, in _create_platform_teams
    return list(
  File "/home/slarse/.repobee/env/lib/python3.9/site-packages/tqdm/std.py", line 1176, in __iter__
    for obj in iterable:
  File "/home/slarse/.repobee/env/lib/python3.9/site-packages/_repobee/command/teams.py", line 21, in create_teams
    existing_teams_dict = {
  File "/home/slarse/.repobee/env/lib/python3.9/site-packages/_repobee/command/teams.py", line 21, in <dictcomp>
    existing_teams_dict = {
  File "/home/slarse/.repobee/env/lib/python3.9/site-packages/_repobee/ext/gitea.py", line 158, in get_teams
    team_name = team_dict["name"]
TypeError: string indices must be integers
```

--- a/src/_repobee/ext/gitea.py
+++ b/src/_repobee/ext/gitea.py
@@ -153,7 +153,14 @@ class GiteaAPI(plug.PlatformAPI):
         team_names = set(team_names or {})
 
         endpoint = f"/orgs/{self._org_name}/teams"
-        response = self._request(requests.get, endpoint)
+        response = self._request(
+            requests.get,
+            endpoint,
+            error_msg=(
+                f"failed to fetch teams from org '{self._org_name}', "
+                "verify that it exists and that token has access!"
+            ),
+        )
         for team_dict in response.json():
             team_name = team_dict["name"]
             if team_names and team_name not in team_names:

--- a/src/_repobee/ext/gitea.py
+++ b/src/_repobee/ext/gitea.py
@@ -156,10 +156,8 @@ class GiteaAPI(plug.PlatformAPI):
         response = self._request(
             requests.get,
             endpoint,
-            error_msg=(
-                f"failed to fetch teams from org '{self._org_name}', "
-                "verify that it exists and that token has access!"
-            ),
+            error_msg="failed to fetch teams from organization: "
+            f"'{self._org_name}'",
         )
         for team_dict in response.json():
             team_name = team_dict["name"]

--- a/system_tests/gitea/test_gitea.py
+++ b/system_tests/gitea/test_gitea.py
@@ -80,6 +80,22 @@ class TestGetTeams:
         # assert
         assert sorted(fetched_team_names) == sorted(team_names + ["Owners"])
 
+    def test_when_target_org_does_not_exist(self):
+        """When the target org does not exist, fetching teams from it should
+        raise an appropriate NotFoundError.
+        """
+        non_existing_org = "some-non-existing-org"
+        api = gitea.GiteaAPI(
+            giteamanager.API_URL,
+            giteamanager.TEACHER_USER,
+            giteamanager.TEACHER_TOKEN,
+            non_existing_org,
+        )
+        with pytest.raises(plug.NotFoundError) as exc_info:
+            list(api.get_teams())
+
+        assert non_existing_org in str(exc_info.value)
+
 
 class TestCreateRepo:
     """Tests for the create_repo function."""

